### PR TITLE
chore: fix dockerfiles now that the submodules have been removed

### LINKED ISSUE #939: Switch default docker compose from fed1 to fed2

### PR DESCRIPTION
**Pull request by o0Ignition0o**
fixes #939

### Update docker-compose and Dockerfiles now that the submodules have been removed ([PR #1950](https://github.com/apollographql/router/pull/1950))

We recently removed git submodules dependency, but we didn't update the global and the fuzzer `docker-compose.yml`.

This PR adds new Dockerfiles and updates `docker-compose.yml` so we can run integration tests and the fuzzer without needing to clone and set up the federation and fed2-demo repositories.

By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/1950

### CODE REVIEW

**Comment by abernix** [comment]
@o0Ignition0o Will this potentially close https://github.com/apollographql/router/issues/939?

**Comment by o0Ignition0o** [comment]
> @o0Ignition0o Will this potentially close #939?

well I didnt see that but i might as well fix it while i m at it!

File: docker-compose.yml [hunk 1]
```diff
@@ -1,15 +1,15 @@
 version: "3.9"
 services:
-
   services:
-    build: dockerfiles/federation-demo
+    build: 
+      dockerfile: dockerfiles/Dockerfile.federation-demo
     ports:
-      - 4100:4100
+      - 4100:4000
```
[the remark this answers is not in the document]
Comment (o0Ignition0o) [reply docker-compose.yml:7]: The docker-compose file was out of date, no container is listening to `4100`. (the federation gateway listens to 4000).

This change binds `localhost:4100` to `services:4000` which won't bind to your host's 4000, so it should be ok ?

File: dockerfiles/Dockerfile.federation-demo [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM node:16-alpine
+
+WORKDIR /src
+RUN apk add git 
+RUN git clone https://github.com/apollographql/federation-demo.git .
+RUN apk del git
+RUN npm install
```
[the remark this answers is not in the document]
Comment (o0Ignition0o) [reply dockerfiles/Dockerfile.federation-demo:7]: gotcha, I separated em when debugging 😅

**Review by Geal — APPROVED** [review]

**Pull request merged**